commit e5af23d811a03cb5e19095b44b801c23bad6744b
Author: Tony Asleson <tasleson@redhat.com>
Date:   Mon Jun 12 16:18:26 2017 -0500

    bcache: Correct return value for sysfs attach errors
    
    If you encounter any errors in bch_cached_dev_attach it will return a negative
    error code.  The variable 'v' which stores the result is unsigned, thus user
    space sees a very large value returned for bytes written which can cause
    incorrect user space behavior.  Utilize 1 signed variable to use throughout
    the function to preserve error return capability.
    
    Signed-off-by: Tony Asleson <tasleson@redhat.com>
    Acked-by: Coly Li <colyli@suse.de>
    Cc: stable@vger.kernel.org

--- a/drivers/md/bcache/sysfs.c
+++ b/drivers/md/bcache/sysfs.c
@@ -192,7 +192,7 @@ STORE(__cached_dev)
 {
 	struct cached_dev *dc = container_of(kobj, struct cached_dev,
 					     disk.kobj);
-	unsigned v = size;
+	ssize_t v = size;
 	struct cache_set *c;
 	struct kobj_uevent_env *env;
 
@@ -227,7 +227,7 @@ STORE(__cached_dev)
 		bch_cached_dev_run(dc);
 
 	if (attr == &sysfs_cache_mode) {
-		ssize_t v = bch_read_string_list(buf, bch_cache_modes + 1);
+		v = bch_read_string_list(buf, bch_cache_modes + 1);
 
 		if (v < 0)
 			return v;
